torch/package/package_exporter.py

```diff
@@ -852,15 +853,89 @@ def _execute_dependency_graph(self):
         extern_file_contents = "\n".join(extern_modules) + "\n"
         self._write(".data/extern_modules", extern_file_contents)
 
+    def _print_debug_info(self):
+        """Print out the debugging information based on the verbosity level."""
+
+        def print_pretty(
+            category: str,
+            entries: List[str],
+            indentation=0,
+            recurse=True,
+            bullet_char="-",
+        ):
+            tab = "    "
+            tabs = tab * indentation
+            print(f"{tabs}{bullet_char} {category}:")
+            tabs += tab
+            for item in sorted(entries):
+                print(f"{tabs}{item}")
+                if recurse and self.verbose > VerbosityLevel.PACKAGE_INFO:
+                    successors = self.dependency_graph._succ[item].keys()
+                    if successors:
+                        print_pretty(
+                            "relies on",
+                            successors,
+                            indentation=2,
+                            recurse=False,
+                            bullet_char="*",
+                        )
+                    predecessors = self.dependency_graph._pred[item].keys()
+                    if predecessors:
+                        print_pretty(
+                            "included in",
+                            predecessors,
+                            indentation=2,
+                            recurse=False,
+                            bullet_char="*",
+                        )
+
+        interned_mods = []
+        explicitly_externed_mods = []
+        implicitly_externed_mods = []
+        mocked_mods = []
+        pickled_objs = []
+        invalid_nodes = []
+        denied_nodes = []
+
+        for module_name, attrs in self.dependency_graph.nodes.items():
```

Comment:
I don't think we should hard code these categories here. Maybe each node should get an additional attribute with what category it should be printed under and we just group nodes by that attribute.